components/Report/Chart/ReportPieChart.tsx

```diff
@@ -0,0 +1,42 @@
+'use client';
+import { Pie, PieChart, Tooltip } from 'recharts';
+import styles from '../Report.module.css';
+
+type ReportPieChartProps = {
+    title: string;
+    data: Array<{ name: string; value: number; fill: string }>;
+};
+
+export default function ReportPieChart(props: ReportPieChartProps) {
+    const { title, data } = props;
+    return (
+        <div className={styles.outerPieChartContainer}>
+            <h2>{title}</h2>
+            <div className={styles.innerPieChartContainer}>
+                <div>
+                    {data.map((dataPoint: { name: string; value: number }) => (
```

Comment:
I think we should consider removing data here with values of 0 from the pie chart so they aren't labeled. Either that or we do it in the outer scope before the data's passed into this chart component, but I can't think of any case where we will want to have a 0 label in the pie chart

Reply:
Also related to this, I do think the text label that shows by default should be the category (e.g. "Male" instead of "8" here) and the on-hover text should be the value. That way the glance value is improved as you can quickly see the ratios of the categories vs the current behavior which shows the ratios and exact values but requires the user to hover to see which category is which.

<img width="222" alt="image" src="https://github.com/Kevinjchang98/St-Francis/assets/17811745/8fc74f64-513e-40e9-b358-96998ec73b4e">

Reply:
great! removed those 0, chart looks clean now!

Reply:
With long label it'll be quite hard to display

![piechart-label](https://github.com/Kevinjchang98/St-Francis/assets/61364158/7adf936b-8b5b-4d55-a2e4-fedfa239a260)

Reply:
Would this look better when shorten the labels, and move each chart to a full width card?

![Uploading shorten-label.png…]()

Reply:
What if you had them side by side but Race being larger than Gender? Is there enough room with the shorter labels? I'm not too sure, what do you think looks best to you?

I think we could remove the % as well to further shorten the labels; the pie chart itself visually represents numeric information but we need categorical information to be easily seen at a glance or if they print the page out or something is my main concern

Reply:
wrapped the label to create some room for side by side container with race container is slightly larger. Small device will be wrapped to full width container. Still a little bit alignment issue with the legend which is quite difficult to customize with recharts.

![inline](https://github.com/Kevinjchang98/St-Francis/assets/61364158/2d85ada1-72e6-4142-b822-4e147632d1c3)